Sources/SwiftDocC/Semantics/Metadata/Metadata.swift

```diff
@@ -132,6 +137,107 @@ public final class Metadata: Semantic, AutomaticDirectiveConvertible {
                 problems.append(Problem(diagnostic: diagnostic, possibleSolutions: [solution]))
             }
         }
+
+        var categorizedAvailability = [MetadataAvailability.Platform : [MetadataAvailability]]()
+        for availability in availability {
+            categorizedAvailability[availability.platform, default: []].append(availability)
+        }
```

Comment:
Minor style preference: there's a dictionary initializer that groups a sequence based on some criteria:

```suggestion
        let categorizedAvailability = Dictionary(grouping: availability, by: \.platform)
```